storage/imx6ull-flash/imx6ull-flash-test.c

```diff
@@ -305,6 +316,122 @@ void test_bootrom(void)
 }
 
 
+/* Checks ECC corrections reported by the BCH */
+void test_ecc(void)
+{
+	const unsigned int paddr = 8 * BLOCK_PAGES_CNT + 1; /* Page under testing (can't contain FCB - different configuration) */
+	const unsigned int rawmetasz = 16 + 26;             /* 16B META + 26B ECC16 */
+	const unsigned int rawdatasz = 512 + 23;            /* 512B DATA + 22.75B ECC14 (adding 2 bits for byte alignment) */
+	const unsigned int maxmetaecc = 16;                 /* Max number of bits per meta chunk ECC can correct */
+	const unsigned int maxdataecc = 14;                 /* Max number of bits per data chunk ECC can correct */
+	unsigned int nblock = paddr / BLOCK_PAGES_CNT;
+	unsigned int start = (paddr % BLOCK_PAGES_CNT) * FLASHDRV_PAGESZ;
+	unsigned int seed = time(NULL);
+	flashdrv_meta_t *aux, *saux;
+	flashdrv_dma_t *dma;
+	uint8_t *data, *sdata;
+	int i, err;
+
+	if ((dma = flashdrv_dmanew()) == MAP_FAILED) {
+		printf("dmanew() failed, err: %d\n", -ENOMEM);
+		return;
+	}
+
+	if ((sdata = mmap(NULL, (FLASHDRV_PAGESZ + _PAGE_SIZE - 1) & ~(_PAGE_SIZE - 1), PROT_READ | PROT_WRITE, MAP_UNCACHED, OID_CONTIGUOUS, 0)) == MAP_FAILED) {
```

Comment:
Maybe instead of using `(FLASHDRV_PAGESZ + _PAGE_SIZE - 1) & ~(_PAGE_SIZE - 1)` ( occurs seven times ) define for example such a variable:
```
const unsigned int maplen = (FLASHDRV_PAGESZ + _PAGE_SIZE - 1) & ~(_PAGE_SIZE - 1);
```
and use it in `mmap()` and `munmap()`.

Reply:
@nalajcie do you agree with this? Maybe we could define PAGE_ALIGN(addr) macro and use it instead? (sidenote: I was instructed not to write such macros before).

Reply:
Added `static size_t pagemapsz` variable storing mapped flash page size. It's initialized in `main()` right after `flashinfo`.